packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,153 @@
+  /**
+   * Id of the Switch button.
+   */
+  switchId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  isChecked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+   * The label text of the Switch component.
+   */
+  labelText: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label.
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Returns Parameters: 1. switch value 2. event.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isChecked: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = (props) => {
+  const {
+    switchId,
+    labelId,
+    isChecked,
+    isDisabled,
+    onChange,
+    labelText,
+    ...customProps
+  } = props;
+  const theme = React.useContext(ThemeContext);
+  const sliderButton = useRef();
+
+  const handleOnClick = useCallback((event) => {
+    // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+    // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+    // This will set focus on the button when clicked.
+    sliderButton.current.focus();
+    if (onChange) {
+      onChange(!isChecked, event);
+    }
+  }, [isChecked, onChange]);
+
+  const handleOnMouseDown = (event) => {
+    event.preventDefault();
+    removeFocusStyles(sliderButton.current);
+  };
+
+  const handleOnKeyDown = (event) => {
+    if (event.nativeEvent.keyCode === KEY_RETURN || event.nativeEvent.keyCode === KEY_SPACE) {
+      enableFocusStyles(sliderButton.current);
+    }
+  };
+
+  const handleOnBlur = () => {
+    restoreFocusStyles(sliderButton.current);
+  };
+
+  const switchClassNames = cx([
+    'switch',
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const trayClassNames = cx([
+    'tray',
+    { 'is-selected': isChecked },
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const sliderClassNames = cx([
+    'slider',
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const statusLabelText = isChecked ? SWITCH_STATE.ON : SWITCH_STATE.OFF;
+
+  const mainClasses = classNames(cx(
+    'switch-wrapper',
+    theme.className,
+  ),
+  customProps.className);
+
+  delete customProps.className;
+  return (
+    <div {...customProps} className={mainClasses}>
+      {/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */}
+      <label
+        htmlFor={switchId}
+        onKeyDown={handleOnKeyDown}
+        onMouseDown={handleOnMouseDown}
+        onBlur={handleOnBlur}
+      >
+        <div className={switchClassNames}>
+          <div className={cx('label-container')}>
+            <div id={labelId} className={cx('label-text')}>{labelText}</div>
+            <div className={cx('status-label-text')}>{statusLabelText}</div>
+          </div>
+          <div className={trayClassNames}>
+            <button
```

Comment:
Should we be using an `<input>` of type `button` here? Looking at the "Accessibility concerns" section of the `<label>` docs raises some questions: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

Have we performed a screen-reader validation of the current implementation to see what it sounds like?

Reply:
This makes me think the whole element should be a button, that way focus doesn't need to be forwarded. That or only the button should have interaction, and the label should be aria hidden, with an aria-label set on the button with the label text. I could see two scenarios:

Internal Label
```
<button
  role="switch"
  aria-checked={value}
  aria-label={labelText}
  tabIndex="0"
  onClick={handleOnClick}
>
  <span aria-hidden="true">{labelText}</span>
  <div>
    {some switch visual styling}
  </div>
</button>
```
External Label
```
<div>
  <span aria-hidden="true">{labelText}</span>
  <button
    role="switch"
    aria-checked={value}
    aria-label={labelText}
    tabIndex="0"
    onClick={handleOnClick}
  >
    <div>
      {some switch visual styling}
    </div>
  </button>
</div>
```

As I believe the current UX guidance has been to have the label selectable as well, I imagine the former might be the desired implementation.

Reply:
> Should we be using an `<input>` of type `button` here? Looking at the "Accessibility concerns" section of the `<label>` docs raises some questions: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label
> 
> Have we performed a screen-reader validation of the current implementation to see what it sounds like?

Currently screen readers picks up the switch interactions for both mouse and keyboard events and reads out Label text and switch status on changing the switch state as shown below : 

![switch-VO](https://user-images.githubusercontent.com/12869809/84771029-56b8f900-aff6-11ea-9a70-12da0b5d7d13.gif)

Reply:
Talking with UX and verifying that the entire region is an interactable element, I would recommend the pattern 2 from above, to make the switch a single element. The unintended affect in the current implementation is that the user has to navigate additional items to reach the switch, and the behavior is inconsistent across browsers, as safari accepts keyboard commands while on a label, but doesn't in chrome. Another is that we end up with a more verbose reading of a single control. Chaning away from the current would also reduce the code complexity as it would no longer need to have focus management.

Reply:
> Talking with UX and verifying that the entire region is an interactable element, I would recommend the pattern 2 from above, to make the switch a single element. The unintended affect in the current implementation is that the user has to navigate additional items to reach the switch, and the behavior is inconsistent across browsers, as safari accepts keyboard commands while on a label, but doesn't in chrome. Another is that we end up with a more verbose reading of a single control. Changing away from the current would also reduce the code complexity as it would no longer need to have focus management.

I think ` the behavior is inconsistent across browsers, as safari accepts keyboard commands while on a label, but doesn't in chrome ` this might be due to outdated deployment iit works same in both safari and Chrome with latest code.

Correct me if I am wrong I don't think if I Remove Label, Slider button will not change on label click and If I wrap label inside button than it will add an empty div inside button to represent slider inside switch-tray and which will again require most of the css to change along with theme variables. 

all the issues that are mentioned above are fixed it is due to travis build deployment link is not updated. not sure how to get deployment link updated to my latest code changes.